mach_eval/analyzers/electromagnetic/flux_linkage_analyzer.py

```diff
@@ -0,0 +1,173 @@
+        print(msg)
+
+        self.app.SetCurrentStudy(0)
+        self.app.GetModel(0).GetStudy(0).DeleteResult()
+
+
+class FluxLinkageJMAG_Analyzer:
+    """Calcuates generates flux linkages from FEA
+
+        Args:
+            problem: object of type Flux_Linkage_Problem holding flux linkage simulation instructions
+        Returns:
+            fea_data: Data dictionary containing information for post-processing
+        """
+    
+    def analyze(self, problem):
+
+        ################################################################
+        # 02. Create all operating points! One per phase + 0 current case
+        ################################################################
+
+        self.study_name = "Machine_FluxLinkage_Study"
+        self.results_filepath = os.path.dirname(__file__) + "/run_data/"
+        # Create output folder
+        if not os.path.isdir(self.results_filepath):
+            os.makedirs(self.results_filepath)
+
+        if not os.path.isdir(self.results_filepath):
+            os.makedirs(self.results_filepath)
+
+        for i in range(len(problem.phase_names)+1):
+
+            # Create transient study with two time step sections
+            study = problem.add_em_study(self.results_filepath, self.study_name)
+
+            # Zero all Currents
+            self.cs_names = problem.zero_currents(study)
+
+            # Set current phase excitation
+            problem.set_currents_sequence(problem.rated_current, study, i, self.cs_names)
+
+            ################################################################
+            # 03. Run electromagnetic studies
+            ################################################################
+            problem.run_study(study, clock_time())
+
+            if i == 0:
+                if os.path.isfile(self.results_filepath + self.study_name + "_flux_of_fem_coil_0.csv") is True:
+                    os.replace(self.results_filepath + self.study_name + "_flux_of_fem_coil.csv", 
+                            self.results_filepath + self.study_name + "_flux_of_fem_coil_0.csv")
+                else:
+                    os.rename(self.results_filepath + self.study_name + "_flux_of_fem_coil.csv", 
+                            self.results_filepath + self.study_name + "_flux_of_fem_coil_0.csv")
+            else:
+                if os.path.isfile(self.results_filepath + self.study_name + "_flux_of_fem_coil_phase_%s.csv" % problem.phase_names[i-1]) is True:
+                    os.replace(self.results_filepath + self.study_name + "_flux_of_fem_coil.csv", 
+                            self.results_filepath + self.study_name + "_flux_of_fem_coil_phase_%s.csv" % problem.phase_names[i-1])
+                else:
+                    os.rename(self.results_filepath + self.study_name + "_flux_of_fem_coil.csv", 
+                            self.results_filepath + self.study_name + "_flux_of_fem_coil_phase_%s.csv" % problem.phase_names[i-1])
```

Comment:
@dmnewman3, I just took a very quick look and I see that this is still using CSV files.

I thought we decided not to use CSV files / or write anything to the hard drive. I.e.:
> problem + analyzer: let's eliminate using CSV files to exchange information. Instead, we'll save flux linkage data to an object. In the problem class, see if you can prevent JMAG from storing the data onto the hard drive. Just retrieve it directly from JMAG via a function call.

Can you implement this?

Reply:
@elsevers , I looked at all of the other analyzers that utilitze JMAG outputs and they all involve writing to and then using .csv files, back to and including the original BSPM analyzer that Barath made. I tried to go into JMAG and see if there was a way around this, but the only manner in which I could generate data was outputting it to a file, csv or otherwise. 

What I changed regarding the use of the .csv files in this newest update was to write to the .csv and then use it's contents to create an object all within the flux linkage analyzer. After this update, we are no longer passing the .csv or it's filepath location between analyzers. This seemed the most logical solution based on our discussion and what I found using JMAG.

Reply:
Okay, if we need to use a CSV file, please keep it contained to happening within the problem class. The analyzer shouldn't have any knowledge of the CSV file existing. 

Remember, this analyzer shouldn't know we're dealing with JMAG. Someone should be able to write a future problem class that this analyzer could use that might use FEMM or Maxwell or ...

Reply:
Ok @elsevers I just updated to move all references of the csv files to the problem class. Please let me know if any further adjustments are necessary.